src/main/java/com/labsynch/labseer/service/MetalotServiceImpl.java

```diff
@@ -313,33 +322,39 @@ public MetalotReturn processAndSave(Metalot metaLot, MetalotReturn mr, ArrayList
 				metalotError = true;
 				throw new DupeParentException("Duplicate parent structure");
 
-			} else if (chemService.checkForSalt(parent.getMolStructure())) {
-				// multiple fragments
-				if (parent.getIsMixture() != null) {
-					if (!parent.getIsMixture()) {
+			} else {
+				Boolean hasMultipleFragments = parent.getMultipleFragments();
+				if (hasMultipleFragments == null){
+					hasMultipleFragments = chemService.checkForSalt(parent.getMolStructure());
+				}
+				if (hasMultipleFragments) {
+					// multiple fragments
```

Comment:
Probably not the responsibility of this PR to fix this but just noting that it seems like the 2 inner if else blocks here can probably be skipped if we add them into line 330 properly.

Reply:
Makes sense. I went ahead and did it, since I'm actively running the salts / fragments test.